### ISSUE TITLE
Unhandled Error  : Process crashed with: TypeError [ERR_INVALID_CHAR]: Invalid character in header content ["Host"]  

### ISSUE THREAD

**Issue by adityathebe** [axios/axios#1804]
```js
const axios = require('axios')

;(async () => {
  const url = "http://www.smun.spb.ru"
  const head_response = await axios.head(url)

  console.log(head_response.headers)
})()
.then(() => { console.log("Success")})
.catch(() => { console.log("Error")})
```

Throws this exception

```bash
Process crashed with: TypeError [ERR_INVALID_CHAR]: Invalid character in header content ["Host"]
    at validateHeader (_http_outgoing.js:506:11)
    at ClientRequest.setHeader (_http_outgoing.js:510:3)
    at new ClientRequest (_http_client.js:228:12)
    at Object.request (http.js:41:10)
    at RedirectableRequest._performRequest (/home/runner/node_modules/follow-redirects/index.js:158:24)
    at RedirectableRequest._processResponse (/home/runner/node_modules/follow-redirects/index.js:249:10)
    at ClientRequest.RedirectableRequest._onNativeResponse (/home/runner/node_modules/follow-redirects/index.js:39:10)
    at Object.onceWrapper (events.js:272:13)
    at ClientRequest.emit (events.js:180:13)
    at HTTPParser.parserOnIncomingClient [as onIncoming] (_http_client.js:540:21)
```

**Comment by chinesedfan** [comment]
It is a problem that the server can't response correctly, instead of an issue of axios. Please look for help in [stackoverflow](https://stackoverflow.com). Seems like your server have redirected the request with something wrong, because the error stack shows in `follow-redirects`.

A quick solution is that checking **url** and **request headers** and comparing with other ways when you can get the right response. If you confirm that axios sends wrong url/headers, feel free to remind me to reopen.

**Comment by Strate** [comment]
@chinesedfan the problem is that incorrect server response crashes server, instead of rejecting promise.

**Comment by chinesedfan** [comment]
Tracking in https://github.com/follow-redirects/follow-redirects/issues/125.

@yasuf @Alanscut And I am curious why `follow-redirects` is locked to 1.5.10 in https://github.com/axios/axios/pull/2570.

**Comment by RubenVerborgh** [comment]
> @chinesedfan the problem is that incorrect server response crashes process, instead of rejecting promise.

`follow-redirects@1.9.1` will now emit such an error as an event, which should cause axios to catch it and reject that promise.

**Comment by QB3L** [comment]
I see current master is using follow-redirects 1.5 so be aware that this is still not fixed

**Comment by namoojaalallee** [comment]
How I could fix it?

**Comment by chinesedfan** [comment]
Closed by #2689.
